src/credentials.rs

```diff
@@ -308,19 +308,19 @@ impl AWSCredentialsProvider for IAMRoleCredentialsProvider {
             let access_key;
             match json_object.find("AccessKeyId") {
                 None => return err("Couldn't find AccessKeyId in response."),
-                Some(val) => access_key = val.to_string().replace("\"", "")
+                Some(val) => access_key = val.as_string().expect("value was not a string").to_owned().replace("\"", "")
```

Comment:
Should we be more specific in these for the `expect` message?  This would say "access key was not a string" or something similar.  Same with the other two below.